Watchman.Discord/Areas/Users/Controllers/WarnsController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Devscord.DiscordFramework.Commons;
+using Devscord.DiscordFramework.Commons.Exceptions;
+using Devscord.DiscordFramework.Commons.Extensions;
+using Devscord.DiscordFramework.Framework.Architecture.Controllers;
+using Devscord.DiscordFramework.Framework.Commands.Parsing.Models;
+using Devscord.DiscordFramework.Framework.Commands.Responses;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
+using Devscord.DiscordFramework.Services.Factories;
+using MongoDB.Bson;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Watchman.Discord.Areas.Protection.Services;
+using Watchman.Discord.Areas.Users.BotCommands;
+using Watchman.Discord.Areas.Users.BotCommands.Warns;
+using Watchman.DomainModel.Messages;
+using Watchman.DomainModel.Users;
+
+namespace Watchman.Discord.Areas.Protection.Controllers
+{
+    public class WarnsController : IController
+    {
+        private readonly MessagesServiceFactory _messagesServiceFactory;
+        private readonly UsersService _usersService;
+        private readonly DirectMessagesService _directMessagesService;
+        private readonly WarnsService _warnService;
+
+        public WarnsController(MessagesServiceFactory messagesServiceFactory, UsersService usersService, DirectMessagesService directMessagesService, WarnsService warnService)
+        {
+            this._messagesServiceFactory = messagesServiceFactory;
+            this._usersService = usersService;
+            this._directMessagesService = directMessagesService;
+            this._warnService = warnService;
+        }
+
+        [AdminCommand]
+        public async Task AddWarn(AddWarnCommand command, Contexts contexts)
+        {
+            var messageService = this._messagesServiceFactory.Create(contexts);
+            var mentionedUser = await this._usersService.GetUserByMentionAsync(contexts.Server, command.User.ToString());
+            await this._warnService.AddWarnToUser(command, contexts, mentionedUser);
+            await messageService.SendResponse(x => x.UserHasBeenWarned(contexts.User.Name, mentionedUser.Name, command.Reason));
+        }
+
+        public Task Warns(WarnsCommand command, Contexts contexts)
+        {
+            var messageService = _messagesServiceFactory.Create(contexts);
+            var mentionedUser = command.User == 0 ? contexts.User : 
+                    this._usersService.GetUserByIdAsync(contexts.Server, command.User).GetAwaiter().GetResult();
+            var warns = this._warnService.GetWarns(command, contexts, mentionedUser, contexts.Server.Id);
```

Comment:
trochę jest to śmiesznie sformatowane. Albo cała instrukcja warunkowa w jednej linii albo tutaj lepiej jakbyś zrobił, że w pierwszej warunek w drugiej dla true i w trzeciej linii dla false kod.

Reply:
ok